File: Data_Structures_and_Algorithms/w4_sorting_and_searching/buble_sort.py
# buble sort algorithm in python


list_1 = [14, 27, 33, 10, 35]

def buble_sort(list_1):
	size = len(list_1)-1

	for i in range(size):
		for j in range(size-i):
			if list_1[j] > list_1[j+1]:
				list_1[j] , list_1[j+1] = list_1[j+1], list_1[j] 


print(list_1)
buble_sort(list_1)
print(list_1)

File: Data_Structures_and_Algorithms/w4_sorting_and_searching/buble_sort.c
// buble sort algorithm in c
#include <stdio.h>

void buble_sort(int list[], int size){
/* sort list */
	int temp;

	for(int i=0; i<size-1; i++){
		for(int s=0; s<size-1-i; s++){

			// exchange variables
			if(list[s]>list[s+1]){
				temp = list[s+1];
				list[s+1] = list[s];
				list[s] = temp;
			}
		}
	}
}

void print_list(int list[], int size){
	//prints list
	for(int i=0; i<size; i++){
		printf("%d ", list[i]);
	}
}

int main(){

	int list[5]={14, 27, 33, 10, 35};

	printf("Before Sorting\n");
	print_list(list, 5);
	buble_sort(list, 5);
	printf("\nAfter Sorting\n");
	print_list(list, 5);


	return 0;
}